opensquirrel/exporter/quantify_scheduler_exporter.py

```diff
@@ -86,21 +82,30 @@ def visit_controlled_gate(self, g: ControlledGate) -> None:
             )
             return
 
-        raise _unsupported_gates_exception
+        raise UnsupportedGateError(
+            f"Cannot export circuit: it contains unsupported gates. The following gate could not be exported: {g}. "
+            "Decompose all gates to the Quantify-scheduler gate set first (rxy, rz, cnot, cz)"
+        )
 
 
 def export(circuit: Circuit) -> "quantify_scheduler.Schedule":
     if "quantify_scheduler" not in globals():
 
         class QuantifySchedulerNotInstalled:
             def __getattr__(self, attr_name: Any) -> None:
-                raise ImportError("quantify-scheduler is not installed, or cannot be installed on your system")
+                raise ModuleNotFoundError("quantify-scheduler is not installed, or cannot be installed on your system")
 
         global quantify_scheduler
         quantify_scheduler = QuantifySchedulerNotInstalled()
         global quantify_scheduler_gates
         quantify_scheduler_gates = QuantifySchedulerNotInstalled()
 
     schedule_creator = _ScheduleCreator(circuit.qubit_register_name)
-    circuit.ir.accept(schedule_creator)
+    try:
+        circuit.ir.accept(schedule_creator)
+    except UnsupportedGateError as e:
+        raise ExportError(
```

Comment:
Not sure about the `ExportError` name:
- I would rather name it `QuantifySchedulerExporterError`.
- `ImportError` is a Python specific error regarding import naming.
- `ExportError` can be confusing in the sense that one would associate it with `ImportError`, when it actually has nothing to do with it, and may even think that it is a Python specific error as well.
- `QuantifySchedulerExporterError` leaves no space to doubt.

Reply:
I would prefer to have the name `QuantifyScheduler` out of the name of the Exception class. What do you think of `CircuitExportError`?

Reply:
I suggested to include `QuantifySchedulerExporter` in the name because this code belongs to Quantify scheduler. But I agree with having a more generic name. I would then use `ExporterError`.

All this naming becomes clearer when you put all error classes together in a file. Then you polish the names looking for consistency.